src/khepri_lock.erl

```diff
@@ -0,0 +1,654 @@
+    _ = case Attempts rem 10 of
+            0 -> rand:seed(exsplus);
+            _ -> ok
+        end,
+    %% First time 1/4 seconds, then doubling each time up to 8 seconds max.
+    Tmax = case Attempts > 5 of
+               true ->
+                   8000;
+               false ->
+                   ((1 bsl Attempts) * 1000) div 8
+           end,
+    T = rand:uniform(Tmax),
+    receive after T -> ok end.
+
+decrement(infinity) ->
+    infinity;
+decrement(N) when is_integer(N) ->
+    N - 1.
+
+-spec group_id_option(options()) -> group_id().
+group_id_option(#{recursive := false, group := _Group} = Options) ->
+    throw({error, {bad_options, Options}});
+group_id_option(#{group := Group}) ->
+    Group;
+group_id_option(#{recursive := true}) ->
+    self();
+group_id_option(_Options) ->
+    erlang:make_ref().
+
+-spec release_on_exit_option(options(), scope()) -> boolean().
+release_on_exit_option(#{release_on_exit := ReleaseOnExit}, _Scope) ->
+    ?assert(is_boolean(ReleaseOnExit)),
+    ReleaseOnExit;
+release_on_exit_option(_Options, local) ->
+    true;
+release_on_exit_option(_Options, cluster) ->
+    false.
+
+-spec release(Lock) -> Ret when
+      Lock :: khepri_lock:lock(),
+      Ret :: ok | error.
+%% @doc Releases the given lock if it exists, synchronously.
+%%
+%% A lock may be held multiple times. This function returns `ok' if the lock
+%% is successfully released once but the lock may still be held - either by the
+%% same process if the lock is recursive or by other processes if the lock is
+%% non-exclusive. See {@link group_id()} for more information.
+%%
+%% @returns `ok' if the lock existed and was released, `error' otherwise.
+
+release(#khepri_lock{scope = local,
+                     store_id = StoreId,
+                     lock_id = LockId,
+                     group_id = GroupId}) ->
+    ResourceId = {StoreId, LockId},
+    Request = {release, ResourceId, GroupId},
+    gen_server:call(?SERVER, Request);
+release(#khepri_lock{scope = cluster} = Lock) ->
+    khepri_machine:release_lock(Lock).
+
+-spec delete(LockId, Scope) -> Ret when
+      LockId :: khepri_lock:lock_id(),
+      Scope :: khepri_lock:scope(),
+      Ret :: khepri:ok(TimesReleased) | khepri:error(),
+      TimesReleased :: non_neg_integer().
+%% @doc Synchronously and forcefully releases all holds on a lock.
+%%
+%% This is the same as calling `delete(StoreId, LockId, Scope)' with the
+%% default store ID (see {@link khepri_cluster:get_default_store_id/0}).
+%%
+%% @see delete/3.
+
+delete(LockId, Scope) ->
+    StoreId = khepri_cluster:get_default_store_id(),
+    delete(StoreId, LockId, Scope).
+
+-spec delete(StoreId, LockId, Scope) -> Ret when
+      StoreId :: khepri:store_id(),
+      LockId :: khepri_lock:lock_id(),
+      Scope :: khepri_lock:scope(),
+      Ret :: khepri:ok(TimesReleased) | khepri:error(),
+      TimesReleased :: non_neg_integer().
+%% @doc Synchronously and forcefully releases all holds on a lock.
```

Comment:
In a past workplace we added a parameter to a "manual" function like this that had to be passed as `42` or else it would crash 😄

I agree, "delete" is too inconspicuous, let's call this "force_release" 👍